### DISCUSSION TITLE
Open sourcing Ezno's checker and trying out / previewing the JavaScript type checker today

### DISCUSSION THREAD

**Discussion by kaleidawave** [kaleidawave/ezno#21] [Announcements]
**Firstly, to not waste anyone's time, do not bother running this currently on any existing projects. It does not currently support `import`, loops, async and loads of other JavaScript features. This is an early release to showcase functions, object, events and basic typing. More of those features coming shortly, once the architecture has been initially tried out.**

## [I have created a quick start guide for trying Ezno today](https://gist.github.com/kaleidawave/5dcb9ec03deef1161ebf0c9d6e4b88d8)

> If you haven't heard of Ezno before you can read the [initial announcement](https://kaleidawave.github.io/posts/introducing-ezno/) for a idea of some features implemented in this release and the aims of the project.

[Ezno's checker is now open source 🎉](https://github.com/kaleidawave/ezno/tree/main/checker). Additionally I have spent the last week integrating it into the [Oxc CLI](https://github.com/Boshen/oxc). If you haven't heard of it before, Oxc is a suite of tools for the JavaScript / TypeScript language and [boshen](https://twitter.com/boshen_c) offered up his fast and correct parser and AST to help get Ezno's checker out quicker. There is now a [new crate](https://github.com/Boshen/oxc/tree/main/crates/oxc_type_synthesis) that combines Oxc's ASTs and Ezno's checker and contexts for fast and correct type checking. While not the end of my own parser, it means I can focus more on type checking rather getting caught up in building a correct JavaScript parser.

You can get the new Oxc CLI [though its release page](https://github.com/Boshen/oxc/releases/tag/v0.1.0), via npm or directly check a file with npx `npx oxidation-compiler@latest check file.ts`. [Be sure to checkout the example file for a walkthrough of currently supported features](https://gist.github.com/kaleidawave/5dcb9ec03deef1161ebf0c9d6e4b88d8).

![image](https://github.com/kaleidawave/ezno/assets/26967284/9b9549ac-0aaf-452b-8fd2-c4b8a25c9e87)

### Next steps

When you find issues, check for duplicates and if not documented before [post them here](https://github.com/kaleidawave/ezno/issues). If you have questions or feedback with the overall direction of the project post them in [discussions](https://github.com/kaleidawave/ezno/discussions). For contributing consider checking out existing issues and the [docs](https://github.com/kaleidawave/ezno/tree/main/checker/docs). In the following weeks I may create a Zulip server for helping contributors.

Support for Ezno is coming soon to [Oxc's online playground](https://boshen.github.io/oxc/playground). I have more updates, posts and features coming over the next few weeks. This release does not include Ezno's own CLI (with the REPL), the LSP and the experimental type based optimisers for both JS and frameworks. They will hopefully be made public in the next few weeks.

You can [follow me to stay up to date on those](https://twitter.com/kaleidawave).

### 🙌

A big thanks for all the have show interest, [early contributors](https://github.com/kaleidawave/ezno/graphs/contributors) who have helped with testing and improving integrations and thanks for [Gregoor](https://github.com/Gregoor), [TomasHubelbauer](https://github.com/TomasHubelbauer) and [reuben](https://github.com/reuben) for sponsoring to help make the project viable! All sponsorships are incredibly valuable, so if you want to see development continued and are in the position to do so, [considering sponsoring today](https://github.com/sponsors/kaleidawave)!

**Comment by joshtriplett** [comment]
Thank you so much for releasing this as Open Source!

**Comment by dimitropoulos** [comment]
would you be interested in coming on [Michigan TypeScript](https://www.youtube.com/@MichiganTypeScript) to do a [showcase ](https://youtu.be/RxMEvD2a26k?t=335)on this?  totally understand if it's too early, but wanted to put the open offer out there for whenever/ifever you're interested.

<details>
<summary>if so...</summary>

the easiest way to connect is probably on [Discord](https://discord.gg/tmhGfR4Dye) or [Twitter](https://twitter.com/MiTypeScript), but you can always email as well (info@michigantypescript.com)

</details>

**Reply by kaleidawave** [reply]
That sounds amazing! Unfortunately I have a lot on ATM and are a bit shy to do video podcasts 😀. Will think about it for later this year (Ezno will hopefully be more stable by then as well!)

**Reply by dimitropoulos** [reply]
sounds great: we're here when you're ready :)

**Comment by guitheengineer** [comment]
Can we have a comparison with TS? Showcasing the advantages over it would be cool to see

**Reply by david-crespo** [reply]
https://kaleidawave.github.io/posts/introducing-ezno/

**Reply by kaleidawave** [reply]
Yes would be good to share differences for people to compare and be aware of the differences. I was drafting up https://github.com/kaleidawave/ezno/blob/main/checker/docs/typescript.md to document it.
